srcs/app/utils/SyscallWrap.cpp

```diff
@@ -1,55 +1,238 @@
 #include <SyscallWrap.hpp>
-#include <fcntl.h>
-#include <sys/wait.h>
-#include <stdlib.h>
-#include <string.h>
-#include <unistd.h>
-#include <cerrno>
-#include <cstdlib>
-#include <cstring>
-#include <stdexcept>
-#include <string>
-
-int SyscallWrap::pipeWr(int pipefd[2]) {
+
+int SyscallWrap::pipeWr(int pipefd[2],
+						const std::string &file,
+						const std::string &func,
+						std::size_t line) {
 	int ret;
 	if ((ret = pipe(pipefd)) == -1) {
-		throw std::runtime_error(std::strerror(errno));;
+		std::ostringstream ss;
+		ss << func << ": " << __FUNCTION__ << ": " << std::strerror(errno)
+		<< ", in " << file << ":"<< line;
```

Comment:
What do you think about extracting this part to a helper method?

Reply:
Done